File: compiled_lang/go/f01.loop.go
// testbox: title="range over slice (index form)"

package main

import (
	"fmt"
	"os"
)

func main() {
	entries, err := os.ReadDir("dirtest")
	if err != nil {
		panic(err)
	}

	// range with only the index, indexing into the slice for the value
	for i := range entries {
		entry := entries[i]
		if entry.IsDir() {
			fmt.Println(entry.Name() + " is a directory")
		} else {
			fmt.Println(entry.Name() + " is not a directory")
		}
	}
}

File: compiled_lang/rust/src/f01.loop.rs
// testbox: title="Iterator::for_each() with closure"
use std::fs;

fn main() {
    let mut entries: Vec<_> = fs::read_dir("dirtest").unwrap().map(|e| e.unwrap()).collect();
    entries.sort_by_key(|e| e.file_name());

    // Iterator::for_each() with a closure
    entries.iter().for_each(|entry| {
        let name = entry.file_name().to_string_lossy().to_string();
        if entry.path().is_dir() {
            println!("{} is a directory", name);
        } else {
            println!("{} is not a directory", name);
        }
    });
}
